addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -391,6 +391,175 @@ options.registry.parallax = options.Class.extend({
+        var self = this;
+
+        var $dialog = new Dialog(null, {
+            title: _t("Facebook page"),
+            $content: $(core.qweb.render("website.facebook_page_dialog", self.fbData)),
+            buttons: [
+                {text: _t("Save"), classes: "btn-primary", close: true, click: function () {
+                    self.$target.empty();
+                    _.each(self.fbData, function (value, key) {
+                        self.$target.attr('data-'+key, value);
+                    });
+                    var actualWidth = self.$target.width();
+                    self.fbData.width = actualWidth > 500 ? 500 : actualWidth < 180 ? 180 : actualWidth;
+                    var iframeSrc = $.param.querystring('https://www.facebook.com/plugins/page.php', self.fbData);
+                    self.$target.append(_.str.sprintf('<iframe src="%s" width="%s" height="%s" style="border:none;overflow:hidden" scrolling="no" frameborder="0" allowTransparency="true"></iframe>', iframeSrc, self.fbData.width, self.fbData.height));
+                }},
+                {text: _t("Discard"), close: true, click: _.bind(this._setParams, this)},
+            ]
+        }).open();
+
+        $dialog.opened().then(function () {
+            $dialog.$footer.find('.btn-primary').prop('disabled', true);
+            $dialog.$el.find('.form-horizontal').on('change', function (e) {
+                // Update values in fbData
+                self.fbData.tabs = _.map($dialog.$('.o_facebook_tabs input:checked'), function (a) {return a.name}).join(',');
+                self.fbData.href = $dialog.$('.o_facebook_page_url').prop('value');
+                _.each($dialog.$('.o_facebook_options input'), function (el) {
+                    self.fbData[el.name] = $(el).prop('checked');
+                });
+                self._renderPreview($dialog);
+            });
+        });
+        if (this.fbData.href) {
+            this._renderPreview($dialog);
+        }
+    },
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Fetch page url from odoo website config.
+     * It required when we Initialize facebook page data.
+     *
+     * @private
+     */
+    _fetchFbUrl: function () {
+        var self = this;
+        return this._rpc({
+            model: 'website',
+            method: 'search_read',
+            args: [[], ['social_facebook']],
+            limit: 1
+        }).then(function (res) {
+            if (res) {
+                self.fbData.href = res[0].social_facebook || 'https://www.facebook.com/Odoo';
+            }
+        });
+    },
+    /**
+     * This method is responsible to display facebook page preview in dialog.
+     * it also verify that URL must be facebook.com|fb.com and it also check that page is exist or not in Facebook.
+     * it Manage height of ifream for better usability.
```

Comment:
Try to avoid useless parts of sentences for documentation comments. We usually start sentences with a conjugated verb. So:
- `This method is responsible to display facebook page preview in dialog` -> `Displays a facebook page preview in the dialog`
- ...

Also try to respect the 80 characters per line limit :wink:

Reply:
Same goes for all other documentation comments in this PR :wink: